libkbfs/block_journal.go

```diff
@@ -362,22 +382,19 @@ func (j *bserverTlfJournal) getAll() (
 	return res, nil
 }
 
-func (j *bserverTlfJournal) putData(
+func (j *blockJournal) putData(
 	id BlockID, context BlockContext, buf []byte,
 	serverHalf BlockCryptKeyServerHalf) error {
 	err := validateBlockServerPut(j.crypto, id, context, buf)
 	if err != nil {
 		return err
 	}
 
-	j.lock.Lock()
-	defer j.lock.Unlock()
-
 	if j.isShutdown {
-		return errBserverTlfJournalShutdown
+		return errBlockJournalShutdown
 	}
 
-	_, existingServerHalf, err := j.getDataLocked(id, context)
+	_, existingServerHalf, err := j.getDataWithContext(id, context)
```

Comment:
Seems a little wasteful to get all the data just to ignore it.  Maybe refactor a bit and add a new function for just checking the context and getting the key half?